### PR TITLE
Import eager mode source tree and submodules

### PR DESCRIPTION
**Pull request by abock**
This PR introduces eager mode support as extracted from https://github.com/microsoft/onnxruntime-pytorch, which represents our work since November (all commits/history is preserved).

To allow us to move forward with integrating both ORTModule and eager mode, I have re-shaped the source tree such that all of the eager mode work lives in `src/eager` for now.

Subsequent PRs will unify the directory tree structure and build, but this should be enough to allow both teams to work out of the same repository as a first step.

### CODE REVIEW

**Comment by jspisak** [comment]
I am curious how you see the workflow for supporting eager mode. Is this not redundant with PyTorch eager mode execution? cc @dzhulgakov

**Comment by abock** [comment]
@jspisak I'm not sure what you mean by redundant. This extension allows us to run individual PyTorch operations through ONNX Runtime's new eager mode support. Individual ops are translated to discrete kernel invocations on the ORT side.

**Comment by dzhulgakov** [comment]
I guess the broader question is: what it'd take to bring the individual operator improvements from ONNX Runtime into upstream PyTorch? Since it's a drop-in replacement both ORT and ATen implementations should have the same semantics. Since it's eager mode, there's no fancy graph optimization going on. So unless there are some crazy 3rd party dependencies it feel that for each op there's one implementation that is strictly better (or a mix of those for different shapes). Upstreaming it to PyTorch would benefit even broader user base. Or am I missing something? (Of course it's a broader discussion than just this PR)

**Comment by souptc** [comment]
@dzhulgakov , there are also many scenarios that people want to run inference in ort without python environment, upstream part of ort kernels to pytorch as a fork will be hard to maintain, and not flexible for the future operations/optimizations. we are hoping with this ort-eager mode api, it will give user a easier way to access the ort optimizations quickly with the unified pytorch eager mode experience, to better integrate in pytorch ecosystem.

**Review by liqunfu — APPROVED** [review]

**Review by SherlockNoMad — APPROVED** [review]

**Review by suffiank — APPROVED** [review]

**Review by asolms — APPROVED** [review]

File: src/eager/test_models/mnist_fc.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+from __future__ import print_function
```
Comment (thiagocrepaldi) [inline src/eager/test_models/mnist_fc.py:1]: I suggest having one main test folder. there can be subfolder to organization, but a single test file so build scripts can handle them easily

**Review by thiagocrepaldi — APPROVED** [review]

File: src/eager/test_models/mnist_fc.py (hunk 1 above)
Comment (abock) [reply src/eager/test_models/mnist_fc.py:1]: 👍 A lot of this still needs cleaning up and organizing, but we want to get the two separate trees (ORTModule + eager) into the same repo to start. I will consolidate tests in a subsequent PR.

File: src/eager/.gitignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+ort_build/
+build/
+ort_aten.g.cpp
```
[the remark this answers is not in the document]
Comment (abock) [reply src/eager/.gitignore:3]: I don't mind moving it, but there is/was great utility in having it side-by-side in the tree. Makes for much easier reviewing of the generated code in the context of `ort_aten.cpp`, which is not generated.

```
src/eager/
├── ort_aten.cpp
└── ort_aten.g.cpp
```

**Review by liqunfu — APPROVED** [review]

**Pull request merged**